spec/unit/mods_pub_fields_spec.rb

```diff
@@ -45,155 +38,220 @@ def sdb_for_mods(m)
+          <dateIssued>2000</dateIssued>" +
+        mods_origin_info_end_str
       end
-      it 'yy00 for yyth century dates' do
-        m = "<mods #{ns_decl}><originInfo>
-              <dateIssued>19th century</dateIssued>
-            </originInfo></mods>"
-        sdb = sdb_for_mods(m)
-        doc_hash = sdb.doc_hash_from_mods
-        expect(doc_hash[:pub_date_sort]).to eq('1800')
+      it 'pub_year_no_approx_isi calls Stanford::Mods::Record instance pub_date_facet_single_value(true)' do
+        sdb = sdb_for_mods(mods)
+        expect(sdb.smods_rec).to receive(:pub_date_facet_single_value).with(true).and_call_original
+        allow(sdb.smods_rec).to receive(:pub_date_facet_single_value).with(false) # for other flavor
+        expect(sdb.doc_hash_from_mods[:pub_year_no_approx_isi]).to eq '2000'
       end
-      it '0y00 for yth century dates' do
-        m = "<mods #{ns_decl}><originInfo>
-              <dateIssued>9th century</dateIssued>
-            </originInfo></mods>"
-        sdb = sdb_for_mods(m)
-        doc_hash = sdb.doc_hash_from_mods
-        expect(doc_hash[:pub_date_sort]).to eq('0800')
+      it 'pub_year_w_approx_isi calls Stanford::Mods::Record instance pub_date_facet_single_value(false)' do
+        sdb = sdb_for_mods(mods)
+        expect(sdb.smods_rec).to receive(:pub_date_facet_single_value).with(false).and_call_original
+        allow(sdb.smods_rec).to receive(:pub_date_facet_single_value).with(true) # for other flavor
+        expect(sdb.doc_hash_from_mods[:pub_year_w_approx_isi]).to eq '1500'
       end
-      it 'works on 3 digit BC dates' do
-        m = "<mods #{ns_decl}><originInfo>
-              <dateCreated>300 B.C.</dateCreated>
-            </originInfo></mods>"
-        sdb = sdb_for_mods(m)
-        doc_hash = sdb.doc_hash_from_mods
-        expect(doc_hash[:pub_date_sort]).to eq('-700')
+      RSpec.shared_examples "single pub year facet" do |field_sym|
+        it_behaves_like 'expected', field_sym, '1945', '1945'
+        it_behaves_like 'expected', field_sym, '945', '945'
+        it_behaves_like 'expected', field_sym, '66', '66'
+        it_behaves_like 'expected', field_sym, '5', '5'
+        it_behaves_like 'expected', field_sym, '0', '0'
+        it_behaves_like 'expected', field_sym, '-4', '4 B.C.'
+        it_behaves_like 'expected', field_sym, '-15', '15 B.C.'
+        it_behaves_like 'expected', field_sym, '-666', '666 B.C.'
+        it_behaves_like 'expected', field_sym, '16--', '17th century'
+        it_behaves_like 'expected', field_sym, '8--', '9th century'
+        it_behaves_like 'expected', field_sym, '19th century', '19th century'
+        it_behaves_like 'expected', field_sym, '9th century', '9th century'
+        it_behaves_like 'expected', field_sym, '300 B.C.', '300 B.C.'
       end
-    end # pub_date_sort
+      it_behaves_like "single pub year facet", :pub_year_no_approx_isi
+      it_behaves_like "single pub year facet", :pub_year_w_approx_isi
+    end
```

Comment:
@peetucket   this is the part esp. for Feigenbaum:  from new 72-108 are the tests for  pub_year_no_approx_isi and pub_year_w_approx_isi, which will be the new pub year facets for Feigenbaum.